datafusion/common/src/scalar.rs

```diff
@@ -3616,6 +3216,19 @@ mod tests {
             Some(vec![Some(4), Some(5)]),
         ]);
         assert_eq!(list_array, &expected);
+
+        let scalars = build_list::<i64>(vec![
+            vec![Some(1), Some(2), Some(3)],
+            vec![Some(4), Some(5)],
+        ]);
```

Comment:
Maybe we can add tests that contain null?

Reply:
I dont think is possible to create list with null? Maybe List(NullArray)? But it is different than List(I64::None). So I did not add it.